lib/src/serializable.dart

```diff
@@ -0,0 +1,224 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library serializable_module.src.serializable;
+
+@MirrorsUsed(metaTargets: 'serializable_module.src.serializable.Reflectable')
+import 'dart:mirrors';
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+import 'package:w_common/disposable.dart' show Disposable;
+import 'package:w_common/json_serializable.dart' show JsonSerializable;
+
+import 'package:w_module/src/event.dart';
+import 'package:w_module/src/module.dart';
+
+// Any classes / methods that are going to be reflected must annotate with this
+class Reflectable {
+  const Reflectable();
+}
+
+abstract class Bridge<T> {
+  final Stream<Map> apiCallReceived;
+  Bridge(this.apiCallReceived);
+  void broadcastSerializedEvent(Map event);
+  void handleSerializedApiCall(T apiCall);
+}
+
+class SerializableEvent<T> extends Event<T> {
+  String _eventKey;
+
+  SerializableEvent(this._eventKey, DispatchKey dispatchKey)
+      : super(dispatchKey);
+
+  String get eventKey => _eventKey;
+}
+
+abstract class SerializableEvents {
+  List<SerializableEvent> get allEvents;
+}
+
+abstract class SerializableModule extends Module {
+  SerializableModule() {
+    SerializableBus.sharedBus.registerModule(this);
+  }
+
+  @override
+  SerializableEvents get events => null;
+
+  String get serializableKey => null;
+}
+
+class _ModuleRegistration extends Object with Disposable {
+  final SerializableModule module;
+
+  _ModuleRegistration(this.module);
+
+  @override
+  void manageStreamSubscription(StreamSubscription subscription) {
+    super.manageStreamSubscription(subscription);
+  }
+}
+
+class SerializableBus {
+  static final SerializableBus sharedBus = new SerializableBus();
+
+  Bridge _bridge;
+  final Logger _logger = new Logger('Serializable Bus');
+  Map<String, _ModuleRegistration> _moduleRegistrations =
+      <String, _ModuleRegistration>{};
+
+  Bridge get bridge => _bridge;
+  set bridge(Bridge bridge) {
+    _bridge = bridge;
+    _bridge.apiCallReceived.listen(_handleApiCall);
+  }
+
+  Map<String, SerializableModule> get registeredModules =>
+      new Map.fromIterable(_moduleRegistrations.keys,
+          value: (key) => _moduleRegistrations[key].module);
+
+  void reset() {
+    _moduleRegistrations.clear();
+    _bridge = null;
```

Comment:
Looking at the [Bridge abstract class](https://github.com/Workiva/w_module/pull/59/files#diff-949ea1a52f226534b3058b4ce26a64edR33), and the [proposed implementation](https://github.com/Workiva/w_viewer_mobile/blob/MOBILE-2307/lib/src/wk_web_view_bridge.dart#L17) : 
We see that there is a `StreamSubscription` created yet never cancelled.

Looking at the reset method, we see that we simply null out the reference to _bridge.  

Two questions: 
1. Should `WKWebViewBridge` support a function to call when we want to clean up resources (aka dispose)?
2. Should `SerializableBus` call that cleanup function prior to nulling out the `_bridge` reference?

One option is to consider using w_common : https://github.com/Workiva/w_common/blob/master/lib/disposable.dart